api-reference/10 UI Widgets/GridBase/1 Configuration/editing/changes.md

```diff
@@ -6,7 +6,27 @@ firedEvents: optionChanged
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+[tags] ctp
+
+Pending row changes.
 
 ---
-<!-- Description goes here -->
\ No newline at end of file
+
+#include common-ctp-note-wo-devextreme 
+
+Each change is an object that can have the following fields:
+
+* `type`: String                 
+Data change type: *"insert"*, *"update"*, or *"remove"*.
+
+* `data`: Object             
+An object with updated row fields.
+
+* `key`: any                     
+The key of the row being updated or removed.
+
+The **changes** array can be changed from the UI, with native JavaScript methods, or with widget methods ([addRow]({basewidgetpath}/Methods/#addRow), [editRow]({basewidgetpath}/Methods/#editRowrowIndex), [editCell]({basewidgetpath}/Methods/#editCellrowIndex_dataField), [deleteRow]({basewidgetpath}/Methods/#deleteRowrowIndex)). However, objects of type *"insert"* cannot be added with native JavaScript methods. We plan to add this functionality in future releases.
```

Comment:
```suggestion
The **changes** array can be changed from the UI, with native JavaScript methods, or with widget methods ([addRow]({basewidgetpath}/Methods/#addRow), [editRow]({basewidgetpath}/Methods/#editRowrowIndex), [editCell]({basewidgetpath}/Methods/#editCellrowIndex_dataField), [deleteRow]({basewidgetpath}/Methods/#deleteRowrowIndex)). However, *"insert"* object types cannot be added with native JavaScript methods. We plan to add this functionality in future releases.
```

Reply:
@arminal 
'objects of type insert', not 'insert object types' is correct. Can it be 'objects of type insert'?